### ISSUE TITLE
Make numbered function defined with `function mydict.f()` syntax have a name by default?

### ISSUE THREAD

**Issue by user202729** [vim/vim#13722]
**Is your feature request about something that is currently impossible or hard to do? Please describe the problem.**

Currently, a numbered function that is defined as a Funcref in a dict will just display as a number in the call stack (/traceback/backtrace/stack trace). This makes it difficult to locate the function.

Of course, `:h numbered-function` suggests typing `:function g:42`, but this is not applicable if the vim instance has already exited and only the log remain e.g. when some user encounter an error traceback, and report the bug. (specific instance is https://github.com/lervag/vimtex/issues/2849 .)

**Describe the solution you'd like**
If a function is defined with the syntax
```
function mydict.f()
    ...
endfunction
```
then the numbered function should somehow get a name that resemble `mydict.f`, or at least `f`.

(Of course I'm aware that later on the user could do something like `let mydict.g=mydict.f` which will make the name incorrect, but the same thing can be done in e.g. Python too, and Python choose to use the original name)

**Comment by chrisbra** [comment]
isn't the number stable between different invocations of Vim? I remember having similar issues with maintaining the airline plugin and I'd recommend to avoid this as much as possible (and such a thing isn't compatible with vim9 script anyhow).

**Comment by user202729** [comment]
Yes, but I think the number depends on which plugin/plugin version the user installs exactly -- in the case of a bug report from an user, it may not be clear.

**Comment by chrisbra** [comment]
yeah, but that's where you have the mentioned `:h numbered-function` which helps you debug this
